packages/arb-token-bridge-ui/src/components/TransferPanel/Routes/OftV2Route.tsx

```diff
@@ -17,28 +17,53 @@ export function OftV2Route() {
   const { isDepositMode } = useNetworksRelationship(networks)
   const { selectedRoute, setSelectedRoute } = useRouteStore()
   const [selectedToken] = useSelectedToken()
-  const sourceChainNativeCurrency = useNativeCurrency({
-    provider: networks.sourceChainProvider
-  })
-  const { feeEstimates, error, isLoading } = useOftV2FeeEstimates({
+
+  const { feeEstimates, error } = useOftV2FeeEstimates({
     sourceChainErc20Address: isDepositMode
       ? selectedToken?.address
       : selectedToken?.l2Address
   })
+  const { estimatedChildChainGasFees, estimatedParentChainGasFees, status } =
+    useGasSummary()
 
   const gasCost = useMemo(() => {
+    if (
+      status !== 'success' ||
+      typeof estimatedParentChainGasFees !== 'number' ||
+      typeof estimatedChildChainGasFees !== 'number'
+    ) {
+      return undefined
+    }
+
     return [
       {
-        gasCost: feeEstimates?.sourceChainGasFee
-          ? feeEstimates.sourceChainGasFee.toString()
-          : undefined,
-        gasToken:
-          'address' in sourceChainNativeCurrency
-            ? sourceChainNativeCurrency
-            : { ...ether, address: constants.AddressZero }
+        gasCost: isDepositMode
+          ? BigNumber.from(
+              (estimatedParentChainGasFees * 1e18).toString()
```

Comment:
Wouldn't it be better to use ethers parsing utils instead of manually multiplying these values?

Reply:
100%, I have updated it